File: src/main/java/org/sgambe/wtplotter/replaydata/Player.java
package org.sgambe.wtplotter.replaydata;

import com.google.gson.JsonObject;

public class Player {
    private String name;
    private String clanTag;
    private int kills;
    private int groundKills;
    private int navalKills;
    private int teamKills;
    private int aiKills;
    private int aiGroundKills;
    private int aiNavalKills;
    private int assists;
    private int deaths;
    private int captureZone;
    private int damageZone;
    private int score;
    private int awardDamage;
    private int missileEvades;
    private String userId;
    private int squadId;
    private int autoSquad;
    private int team;

    public Player fromJson(JsonObject json){
        this.name = json.get("name").getAsString();
        this.clanTag = json.get("clanTag").getAsString();
        this.kills = json.get("kills").getAsInt();
        this.groundKills = json.get("groundKills").getAsInt();
        this.navalKills = json.get("navalKills").getAsInt();
        this.teamKills = json.get("teamKills").getAsInt();
        this.aiKills = json.get("aiKills").getAsInt();
        this.aiGroundKills = json.get("aiGroundKills").getAsInt();
        this.aiNavalKills = json.get("aiNavalKills").getAsInt();
        this.assists = json.get("assists").getAsInt();
        this.deaths = json.get("deaths").getAsInt();
        this.captureZone = json.get("captureZone").getAsInt();
        this.damageZone = json.get("damageZone").getAsInt();
        this.score = json.get("score").getAsInt();
        this.awardDamage = json.get("awardDamage").getAsInt();
        this.missileEvades = json.get("missileEvades").getAsInt();
        this.userId = json.get("userId").getAsString();
        this.squadId = json.get("squadId").getAsInt(); //- (this.team == 1 ? 1000 : 2000);
        this.autoSquad = json.get("autoSquad").getAsInt();
        this.team = json.get("team").getAsInt();
        return this;
    }

    @Override
    public String toString(){
        return "Player{" +
                "name='" + name + '\'' +
                ", clanTag='" + clanTag + '\'' +
                ", kills=" + kills +
                ", groundKills=" + groundKills +
                ", navalKills=" + navalKills +
                ", teamKills=" + teamKills +
                ", aiKills=" + aiKills +
                ", aiGroundKills=" + aiGroundKills +
                ", aiNavalKills=" + aiNavalKills +
                ", assists=" + assists +
                ", deaths=" + deaths +
                ", captureZone=" + captureZone +
                ", damageZone=" + damageZone +
                ", score=" + score +
                ", awardDamage=" + awardDamage +
                ", missileEvades=" + missileEvades +
                ", userId='" + userId + '\'' +
                ", squadId=" + squadId +
                ", autoSquad=" + autoSquad +
                ", team=" + team +
                '}';
    }

    public String getName() {
        return name;
    }

    public String getClanTag() {
        return clanTag;
    }

    public int getKills() {
        return kills;
    }

    public int getGroundKills() {
        return groundKills;
    }

    public int getNavalKills() {
        return navalKills;
    }

    public int getTeamKills() {
        return teamKills;
    }

    public int getAiKills() {
        return aiKills;
    }

    public int getAiGroundKills() {
        return aiGroundKills;
    }

    public int getAiNavalKills() {
        return aiNavalKills;
    }

    public int getAssists() {
        return assists;
    }

    public int getDeaths() {
        return deaths;
    }

    public int getCaptureZone() {
        return captureZone;
    }

    public int getDamageZone() {
        return damageZone;
    }

    public int getScore() {
        return score;
    }

    public int getAwardDamage() {
        return awardDamage;
    }

    public int getMissileEvades() {
        return missileEvades;
    }

    public String getUserId() {
        return userId;
    }

    public int getSquadId() {
        return squadId;
    }

    public int getAutoSquad() {
        return autoSquad;
    }

    public int getTeam() {
        return team;
    }

    // Getters and setters
}

File: classes/player.cpp
#include "player.h"

// Constructor
Player::Player()
    : kills(0), groundKills(0), navalKills(0), teamKills(0),
    aiKills(0), aiGroundKills(0), aiNavalKills(0), assists(0),
    deaths(0), captureZone(0), damageZone(0), score(0),
    awardDamage(0), missileEvades(0), squadId(0), autoSquad(0), team(0) {}

Player Player::fromJson(const QJsonObject &json) {
    Player player;

    if (json.contains("name") && json["name"].isString()) {
        player.name = json["name"].toString();
    }
    if (json.contains("clanTag") && json["clanTag"].isString()) {
        player.clanTag = json["clanTag"].toString();
    }
    if (json.contains("kills") && json["kills"].isDouble()) {
        player.kills = json["kills"].toInt();
    }
    if (json.contains("groundKills") && json["groundKills"].isDouble()) {
        player.groundKills = json["groundKills"].toInt();
    }
    if (json.contains("navalKills") && json["navalKills"].isDouble()) {
        player.navalKills = json["navalKills"].toInt();
    }
    if (json.contains("teamKills") && json["teamKills"].isDouble()) {
        player.teamKills = json["teamKills"].toInt();
    }
    if (json.contains("aiKills") && json["aiKills"].isDouble()) {
        player.aiKills = json["aiKills"].toInt();
    }
    if (json.contains("aiGroundKills") && json["aiGroundKills"].isDouble()) {
        player.aiGroundKills = json["aiGroundKills"].toInt();
    }
    if (json.contains("aiNavalKills") && json["aiNavalKills"].isDouble()) {
        player.aiNavalKills = json["aiNavalKills"].toInt();
    }
    if (json.contains("assists") && json["assists"].isDouble()) {
        player.assists = json["assists"].toInt();
    }
    if (json.contains("deaths") && json["deaths"].isDouble()) {
        player.deaths = json["deaths"].toInt();
    }
    if (json.contains("captureZone") && json["captureZone"].isDouble()) {
        player.captureZone = json["captureZone"].toInt();
    }
    if (json.contains("damageZone") && json["damageZone"].isDouble()) {
        player.damageZone = json["damageZone"].toInt();
    }
    if (json.contains("score") && json["score"].isDouble()) {
        player.score = json["score"].toInt();
    }
    if (json.contains("awardDamage") && json["awardDamage"].isDouble()) {
        player.awardDamage = json["awardDamage"].toInt();
    }
    if (json.contains("missileEvades") && json["missileEvades"].isDouble()) {
        player.missileEvades = json["missileEvades"].toInt();
    }
    if (json.contains("userId") && json["userId"].isString()) {
        player.userId = json["userId"].toString();
    }
    if (json.contains("squadId") && json["squadId"].isDouble()) {
        player.squadId = json["squadId"].toInt();
    }
    if (json.contains("autoSquad") && json["autoSquad"].isDouble()) {
        player.autoSquad = json["autoSquad"].toInt();
    }
    if (json.contains("team") && json["team"].isDouble()) {
        player.team = json["team"].toInt();
    }

    return player;
}

QString Player::getName() const { return name; }
QString Player::getClanTag() const { return clanTag; }
int Player::getKills() const { return kills; }
int Player::getGroundKills() const { return groundKills; }
int Player::getNavalKills() const { return navalKills; }
int Player::getTeamKills() const { return teamKills; }
int Player::getAiKills() const { return aiKills; }
int Player::getAiGroundKills() const { return aiGroundKills; }
int Player::getAiNavalKills() const { return aiNavalKills; }
int Player::getAssists() const { return assists; }
int Player::getDeaths() const { return deaths; }
int Player::getCaptureZone() const { return captureZone; }
int Player::getDamageZone() const { return damageZone; }
int Player::getScore() const { return score; }
int Player::getAwardDamage() const { return awardDamage; }
int Player::getMissileEvades() const { return missileEvades; }
QString Player::getUserId() const { return userId; }
int Player::getSquadId() const { return squadId; }
int Player::getAutoSquad() const { return autoSquad; }
int Player::getTeam() const { return team; }
